### PR TITLE
Chore: Minor tweaks to the documentation and snippets

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request some updates were made to the snippets to make them more clear and some pages in the documentation were updated.

What was done:
- Wrong naming of story files are fixed.
- Added missing component/stories mentions in the snippets.
- Snippets for React typescript files are now properly named, instead of being mentioned as `Button.stories.ts`, they are now called `Button.stories.tsx` in the commented block under each one.
- Added necessary imports to make the snippets more clear.
- Removed `common/button-story-disable-docspage-component.ts.mdx` file and mention from the Docs page in the Writing Docs section, as it was actually unnecessary to get the message across in the documentation.
- Properly named the snippet used in Story layout page in the Configure section and fixed the reference. 

Feel free to provide feedback

### CODE REVIEW

File: docs/snippets/common/other-foo-bar-story.js.mdx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ```js
-// your-story.story.js
+// Foo-Bar.stories.js
```
Comment (domyen) [inline docs/snippets/common/other-foo-bar-story.js.mdx:2]:
```suggestion
// FooBar.stories.js
```

File: docs/snippets/react/button-story-component-decorator.ts.mdx [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ```ts
-// button.stories.ts
-import React from 'react'
-import { Story, Meta } from '@storybook/react/types-6-0';
+// button.stories.tsx
```
Comment (domyen) [inline docs/snippets/react/button-story-component-decorator.ts.mdx:2]:
```suggestion
// Button.stories.tsx
```

File: docs/snippets/react/page-story-with-args-composition.ts.mdx [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // your-page.stories.js
```
Comment (domyen) [inline docs/snippets/react/page-story-with-args-composition.ts.mdx:2]:
```suggestion
// YourPage.stories.js
```

Might as well do camelcase consistently

File: docs/snippets/react/page-story.ts.mdx [hunk 4]
```diff
@@ -1,4 +1,6 @@
 ```ts
+// Page.stories.tsx
```
Comment (domyen) [inline docs/snippets/react/page-story.ts.mdx:2]:
```suggestion
// Page.stories.tsx

```

File: docs/snippets/react/simple-page-implementation.ts.mdx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ```js
-// your-page.ts
+// your-page.tsx
```
Comment (domyen) [inline docs/snippets/react/simple-page-implementation.ts.mdx:2]:
```suggestion
// YourPage.tsx
```

File: docs/snippets/react/your-component-with-decorator.ts.mdx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 ```ts
-// your-component.stories.ts
+// your-component.stories.tsx
```
Comment (domyen) [inline docs/snippets/react/your-component-with-decorator.ts.mdx:2]:
```suggestion
// YourComponent.stories.tsx
```

I notice we use YourComponent and MyComponent in these snippets. Should we standardize on one pattern?

**Review by domyen — CHANGES_REQUESTED** [review]

**Comment by tmeasday** [comment]
OK by me

File: docs/snippets/react/simple-page-implementation.ts.mdx (hunk 5 above)
Comment (jonniebigodes) [reply docs/snippets/react/simple-page-implementation.ts.mdx]: Fixed with the latest commit

File: docs/snippets/react/your-component-with-decorator.ts.mdx (hunk 6 above)
Comment (jonniebigodes) [reply docs/snippets/react/your-component-with-decorator.ts.mdx]: Fixed with the latest commit. @domyen . That's something that i need to take care of, I'm going to do a search on the entire documentation and see if we can consolidate some of the snippets and manage them better.

File: docs/snippets/react/page-story.ts.mdx (hunk 4 above)
Comment (jonniebigodes) [reply docs/snippets/react/page-story.ts.mdx:2]: Fixed with the latest commit

File: docs/snippets/react/page-story-with-args-composition.ts.mdx (hunk 3 above)
Comment (jonniebigodes) [reply docs/snippets/react/page-story-with-args-composition.ts.mdx]: Fixed with the latest commit.

File: docs/snippets/react/button-story-component-decorator.ts.mdx (hunk 2 above)
Comment (jonniebigodes) [reply docs/snippets/react/button-story-component-decorator.ts.mdx]: Fixed with the latest commit

File: docs/snippets/common/other-foo-bar-story.js.mdx (hunk 1 above)
Comment (jonniebigodes) [reply docs/snippets/common/other-foo-bar-story.js.mdx]: Fixed with the latest commit

**Comment by jonniebigodes** [comment]
With the latest commit the feedback was addressed and some other files containing snippets were fixed as well. 

I'm going to check the conflicts and fix them so that the pull request can be merged

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**